### PR TITLE
Add lamb@0.51.0 w/ git auto-update

### LINKED ISSUE #10675: [Request] Add lamb
**Library name:** lamb
**Git repository url:** https://github.com/ascartabelli/lamb
**npm package name or url** (if there is one): https://www.npmjs.com/package/lamb
**License (List them all if it's multiple):** MIT
**Official homepage:**  https://ascartabelli.github.io/lamb/

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀 .

If you'll like to work locally, I'll suggest you start with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are too many files in this repo, without sparseCheckout, the whole process will be super slow, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs#adding-a-new-or-updating-an-existing-library
 - https://github.com/cdnjs/cdnjs/blob/master/.github/CONTRIBUTING.md#d-adding-a-new-library

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄 .

### PR DESCRIPTION
**Pull request by progietheus**
Resolves #10675
[Request] Add lamb

Pull request for issue: #10675

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/ascartabelli/lamb.git
 * Official website (optional, not the repository): https://ascartabelli.github.io/lamb/index.html
 * NPM package url (optional): https://www.npmjs.com/package/lamb
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 3
   - Count of stars: 11
   - Count of forks: 5
 * NPM download stats (optional):
   - Downloads in the last day: 3
   - Downloads in the last week: 396
   - Downloads in the last month: 831

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/lamb/0.51.0/lamb.min.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+//# sourceMappingURL=lamb.min.js.map
```
Comment (pvnr0082t) [inline ajax/libs/lamb/0.51.0/lamb.min.js]: @progietheus Please commit `lamb.min.js.map` also. Thanks

File: ajax/libs/lamb/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "javascript"
+  ],
+  "license": "MIT", 
+  "name": "lamb", 
+  "repository": { 
+    "type": "git",
+    "url": "https://github.com/ascartabelli/lamb.git"
+  },
+  "version": "0.51.0", 
+  "dependencies": {
```
Comment (pvnr0082t) [inline ajax/libs/lamb/package.json]: Let's remove this empty field. Thanks.

File: ajax/libs/lamb/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  "version": "0.51.0", 
+  "dependencies": { 
+    
+  },
+"autoupdate": { 
+    "source": "git",
+    "target": "git://github.com/ascartabelli/lamb.git",
+    "fileMap": [
+      {
+        "basePath": "",
```
Comment (pvnr0082t) [inline ajax/libs/lamb/package.json]: We should fetch files under `dist` in this case, so let's use `dist` as basePath.

File: ajax/libs/lamb/package.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        "files": [
+          "lamb*.js"
```
Comment (pvnr0082t) [inline ajax/libs/lamb/package.json]: We can use powerful rule `"**/*"` instead. Thanks.

**Comment by progietheus** [comment]
Thank you for reviewing my pull request! I will go over your revisions as soon as possible.

File: ajax/libs/lamb/package.json (hunk 3 above)
Comment (progietheus) [inline ajax/libs/lamb/package.json]: I was wondering about this, is the basePath always `dist`?

**Comment by progietheus** [comment]
@pvnr0082t Thanks for the suggestion on using `tools/fixFormat.js`. I will have to use this method next time, possibly sooner than later if my manual fixes do not suffice. Thank you very much for reviewing my PR! I am submitting a new PR now.

**Comment by progietheus** [comment]
I guess I'll have to run the fixFormat utility as my manual fixes, don't seem to be doing the trick. Thing is I am not at home, I'll see about configuring a new environment for tonight.

**Comment by progietheus** [comment]
@pvnr0082t How do you guys run the fixFormat utility? Node?

**Comment by pvnr0082t** [comment]
@progietheus Yeah, you should have node and npm. Run `npm install` at the root of cdnjs to install the necessary dependencies. Then, try `./tools/fixFormat.js`.

**Comment by maruilian11** [comment]
Hi @progietheus, the error msg comes `The branch lamb for this pull request is too old, please rebase this branch with the latest master branch from upstream!`.
If you use the command line, would you please rebase the branch with our latest master? Or if you use website GUI, I'm willing to rebase it for you :smiley:

**Comment by progietheus** [comment]
@maruilian11 Hi, I apologize for the delay in getting this PR resolved. I am just having issues trying to rebase the branch due to the size of the repo. My git status is taking a few minutes even to return anything, I am going to try and resolve this, but if I can't then you can just rebase. I will get back to you. Thank you!

**Comment by maruilian11** [comment]
hi @progietheus  I think this PR is ok now, Thank you for your great contribution! Now we only need to squash these commits into one clear commit, would you like to do it yourself? Or I'm willing to help you do that.

**Comment by progietheus** [comment]
@maruilian11 I'll give it a shot, I was just trying to do this!

**Comment by progietheus** [comment]
Whoops one last fix and I think I'm done

File: ajax/libs/lamb/0.51.0/lamb.min.js (hunk 1 above)
Comment (progietheus) [inline ajax/libs/lamb/0.51.0/lamb.min.js]: I have submitted lamb.min.js.map

**Comment by progietheus** [comment]
@maruilian11 Sorry, but can you explain what to do? I am a little confused.

**Comment by maruilian11** [comment]
@progietheus Yes, just make the commit msg like this:
```
Add lamb@0.51.0 w/ git auto-update,

close #10675, cc @ascartabelli
```

You can just use `git ci --amend` to modify the commit msg.
And please rebase the branch with our latest master, too.
Feel free to ask if you need any help.

**Comment by progietheus** [comment]
Thank you!

-John M. Balzac
balzac.john@gmail.com
(787)503-6824

**Comment by ascartabelli** [comment]
Well, maybe unintended, but "cc" received.

Thank you for including Lamb and for your work!

**Comment by progietheus** [comment]
@maruilian11 I think everything checks out now, please let me know if there is anything you need from my end! Thank you!

**Comment by PeterDaveHello** [comment]
Thanks all 👍

**Comment by ascartabelli** [comment]
Thanks @progietheus for doing the heavy lifting!

Will add a link to cdnjs in the Lamb's home page and readme in the next release.

**Comment by PeterDaveHello** [comment]
Thanks @ascartabelli

**Comment by progietheus** [comment]
My pleasure! Thank you 😊.

-John M. Balzac
balzac.john@gmail.com
(787)503-6824

**Pull request merged**